enterprise/server/backends/userdb/userdb.go

```diff
@@ -579,7 +583,29 @@ func (d *UserDB) createUser(ctx context.Context, tx *db.DB, u *tables.User) erro
 	}
 
 	for _, groupID := range groupIDs {
-		err := tx.Exec("INSERT INTO UserGroups (user_user_id, group_group_id, membership_status) VALUES(?, ?, ?)", u.UserID, groupID, int32(grpb.GroupMembershipStatus_MEMBER)).Error
+		err := tx.Exec(`
+			INSERT INTO UserGroups (user_user_id, group_group_id, membership_status, role)
+			VALUES(?, ?, ?, ?)
+			`, u.UserID, groupID, int32(grpb.GroupMembershipStatus_MEMBER), perms.DefaultRole,
+		).Error
+		if err != nil {
+			return err
+		}
+		// Promote from default role to admin if the user is the only one in the
+		// group after joining.
+		err = tx.Exec(`
+			UPDATE UserGroups
+			SET role = ?
+			WHERE group_group_id = ?
+			AND NOT EXISTS (
+				SELECT * FROM (
+					SELECT * FROM UserGroups
+					WHERE group_group_id = ?
+					AND user_user_id != ?
+				) AS PreExistingUserGroups
```

Comment:
it feels like there is an extra "layer" of "select *" here -- is that needed?

Reply:
This is a workaround for this issue - https://stackoverflow.com/questions/45494/mysql-error-1093-cant-specify-target-table-for-update-in-from-clause

It's probably simpler to write this as 2 separate queries though, which I *think* should be OK since this is a transaction. (Done)